app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1233,9 +1233,8 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = false
     )
 
-    private val homescreenSections: Map<HomeScreenSection, Boolean> by lazy {
+    private val homescreenSections: Map<HomeScreenSection, Boolean> get() =
```

Comment:
When this was implemented Nimbus didn't have a layer of caching, for this reason we made lazy.

Reply:
As the value was lazy and we introduced `onNimbusStartupAndUpdate` very earlier when nimbus was not ready, we were caching the first value and never updating it again.